crates/ffmpeg/src/io/input.rs

```diff
@@ -146,3 +146,125 @@ impl Input<()> {
+mod tests {
+    use std::io::Cursor;
+
+    use super::{FfmpegError, Input, InputOptions, DEFAULT_BUFFER_SIZE};
+
+    #[test]
+    fn test_input_options_default() {
+        let default_options = InputOptions::default();
+
+        assert_eq!(default_options.buffer_size, DEFAULT_BUFFER_SIZE);
+        assert!(default_options.dictionary.is_empty());
+        assert!(default_options.interrupt_callback.is_none());
+    }
+
+    #[test]
+    fn test_open_valid_file() {
+        let valid_file_path = "../../assets/avc_aac_large.mp4";
+        assert!(std::path::Path::new(valid_file_path).exists(), "Test file does not exist");
+
+        let result = Input::open(valid_file_path);
+        assert!(result.is_ok(), "Expected success but got error");
+    }
+
+    #[test]
+    fn test_open_invalid_path() {
+        let invalid_path = "invalid_file.mp4";
+        let result = Input::open(invalid_path);
+        assert!(result.is_err(), "Expected an error for invalid path");
+        if let Err(err) = result {
+            match err {
+                FfmpegError::Code(_) => (),
+                _ => panic!("Unexpected error type: {:?}", err),
+            }
+        }
+    }
+
+    #[test]
+    fn test_new_with_default_options() {
+        let valid_media_data: Vec<u8> = include_bytes!("../../../../assets/avc_aac_large.mp4").to_vec();
+        let data = Cursor::new(valid_media_data);
+        let result = Input::new(data);
+
+        if let Err(e) = &result {
+            eprintln!("Error encountered: {:?}", e);
+        }
+
+        assert!(result.is_ok(), "Expected success but got error");
+    }
+
+    #[test]
+    fn test_seekable_with_valid_input() {
+        let valid_media_data: Vec<u8> = include_bytes!("../../../../assets/avc_aac_large.mp4").to_vec();
+        let data = Cursor::new(valid_media_data);
+        let result = Input::seekable(data);
+
+        if let Err(e) = &result {
+            eprintln!("Error encountered: {:?}", e);
+        }
+
+        assert!(result.is_ok(), "Expected success but got error");
+    }
+
+    #[test]
+    fn test_as_ptr() {
+        let valid_file_path = "../../assets/avc_aac_large.mp4";
+        let input = Input::open(valid_file_path).expect("Failed to open valid file");
+
+        let ptr = input.as_ptr();
+        assert!(!ptr.is_null(), "Expected non-null pointer");
+    }
+
+    #[test]
+    fn test_as_mut_ptr() {
+        let valid_file_path = "../../assets/avc_aac_large.mp4";
+        let mut input = Input::open(valid_file_path).expect("Failed to open valid file");
+
+        let ptr = input.as_mut_ptr();
+        assert!(!ptr.is_null(), "Expected non-null mutable pointer");
+    }
+
+    #[test]
+    fn test_streams() {
+        let valid_file_path = "../../assets/avc_aac_large.mp4";
+        let input = Input::open(valid_file_path).expect("Failed to open valid file");
+
+        let streams = input.streams();
+        assert!(streams.len() > 0, "Expected at least one stream");
+    }
```

Comment:
I think we should assert the actual underlying streams. 
We should check how many streams this has and that they are what we expect when decoding the avc_aac_large.mp4 file.

You can do this when you use `insta::assert_debug_snapshot` use the inline format
```
insta::assert_debug_snapshot(input.streams.collect::<Vec<_>>, @"")
```

So we can make sure the state does not change if it is correct now.